---
title: "Final Project Data Processing Steps"
author: "Alyssa Monda"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# knitr chunk options set to prevent
# code, warnings and messages from being
# shown in your final document
knitr::opts_chunk$set(echo = FALSE)
knitr::opts_chunk$set(warning = FALSE)
knitr::opts_chunk$set(message = FALSE)

#Downloaded packages 
library(phyloseq)
library(dada2)
library(ggplot2)
library(biom)
library(devtools)
library(knitr)
library(mixOmics)
library(tidyverse)
library(vegan)
library(mixOmics)
library(readxl)
library(tableone)
library(ReporteRs)
library(magrittr)
library(plyr)


#If error in phyloseq load visit this page <https://joey711.github.io/phyloseq/install.html>
```

##Import Meta Data
Import Meta Data as DataFrame
```{r}
library(readr)
Meta <- read_delim("~/Desktop/ag_1k_fecal.txt", 
    "\t", escape_double = FALSE, col_types = cols(AGE_YEARS = col_number(), 
        BMI = col_number(), `ENA-BASE-COUNT` = col_number(), 
        HEIGHT_CM = col_number(), VIOSCREEN_FIBER = col_number(),
        VIOSCREEN_LOW_FAT_DAIRY_SERVING = col_number(), 
        VIOSCREEN_LYSINE = col_number(), 
        VIOSCREEN_MANNITOL = col_number(), 
        VIOSCREEN_SFA100 = col_double(), #capric acid
        VIOSCREEN_SFA40 = col_double(),#SFA4:0 Butyric Acid
        VIOSCREEN_SFA170 = col_double(), #margaric acid
        VIOSCREEN_VEG5_DAY = col_number(), 
        VIOSCREEN_VEGETABLE_SERVINGS = col_number(), 
        chao1_1k = col_double()), trim_ws = TRUE)
View(Meta)
```

##Analysis of MetaData
It is important to clean the metadata prior to merging it with the data from the biom file. Therefore I will now recode, clean, and analyze the demographics of the sample. Before I can start I must recategorize the variables and assign values to groups I am interested in.

I will also subset the data to only include individuals over the age of 40. I am using older individuals in this analysis because this is the population I will be working with for my dissertation, and i would love to be able to use this information as preliminary data. However, murine models show differences in the response to exercise based on age, showing juveline mice have increased diversity after an exercise intervention when compared to older mice after the same intervention. I would like to do a future analysis on children and teens after finishing this project. 

```{r}

#Recode labels and levels for current variables to clean up data
#Age
Meta$Age<-Meta$"AGE_YEARS"
Age<-"AGE_YEARS"
#Subset by age after recording at the end. 

#BMI- coded as wanted check implausible values after recoding other variables

#Sex 
Meta$Sex<-Meta$"SEX"
Meta$Sex[Meta$Sex=="male"] <- "Male"
Meta$Sex[Meta$Sex=="female"] <- "Female"
Meta$Sex[Meta$Sex=="other"] <- "Other"
Meta$Sex[Meta$Sex=="Unknown"] <- "NA"
Meta$Sex[Meta$Sex=="unspecified"] <- "NA"

#Race 
Meta$Race<- Meta$"RACE"
Meta$Race[Meta$Race=="Caucasian"] <- "White"
Meta$Race[Meta$Race=="African American"] <- "Black"
Meta$Race[Meta$Race=="Hispanic"] <- "Hispanic"
Meta$Race[Meta$Race=="Asian or Pacific Islander"] <- "Other"
Meta$Race[Meta$Race=="Other"] <- "Other"
Meta$Race[Meta$Race=="Unspecified"] <- "NA"
Meta$Race[Meta$Race=="Unknown"] <- "NA"

#Education 
Education<- "LEVEL_OF_EDUCATION"
Meta$Education<- Meta$"LEVEL_OF_EDUCATION"
Meta$Education[Meta$Education=="Did not complete high school"] <- "< High School Degree"
Meta$Education[Meta$Education=="High School or GED equivalent degree"] <- "High School Degree"
Meta$Education[Meta$Education=="Some college or technical school"] <- "High School Degree"
Meta$Education[Meta$Education=="Associate's degree"] <- "Undergraduate degree"
Meta$Education[Meta$Education=="Bachelor's degree"] <- "Undergraduate degree"
Meta$Education[Meta$Education=="Some graduate school or professional"] <- "Undergraduate degree"
Meta$Education[Meta$Education=="Graduate or Professional degree"] <- "Graduate or Professional degree"
Meta$Education[Meta$Education=="Uknown"] <- "NA"
Meta$Education[Meta$Education=="Unspecified"] <- "NA"

#Do I want to include country of birth, or economic region?

#Antibiotics 
Meta$Antibiotics<-Meta$"ANTIBIOTIC_HISTORY"
Meta$Antibiotics[Meta$Antibiotics=="6 months"] <- "6 Months"
Meta$Antibiotics[Meta$Antibiotics=="Uknown"] <- "> 1 year"
Meta$Antibiotics[Meta$Antibiotics=="Uknown"] <- "NA"
Meta$Antibiotics[Meta$Antibiotics=="Unspecified"] <- "NA"

#Probiotics
Meta$Probiotics<- Meta$"PROBIOTIC_FREQUENCY"
Meta$Probiotics[Meta$Probiotics=="Unknown"] <- "NA"
Meta$Probiotics[Meta$Probiotics=="Unspecified"] <- "NA"

#ExFreq 
ExFreq<- "EXERCISE_FREQUENCY"
Meta$ExFreq <-Meta$"EXERCISE_FREQUENCY"
Meta$ExFreq[Meta$ExFreq=="Unknown"] <- "NA"
Meta$ExFreq[Meta$ExFreq=="Unspecified"] <- "NA"
Meta$ExFreq[Meta$ExFreq=="unknown"] <- "NA"
Meta$ExFreq[Meta$ExFreq=="unspecified"] <- "NA"

#ExLoc 
Meta$ExLoc<- Meta$"EXERCISE_LOCATION"
Meta$ExLoc[Meta$ExLoc=="Depends on the season"] <- "Both"
Meta$ExLoc[Meta$ExLoc=="Unknown"] <- "NA"
Meta$ExLoc[Meta$ExLoc=="Unspecified"] <- "NA"

#PoolFreq 
Meta$PoolFreq<- Meta$"POOL_FREQUENCY"
Meta$PoolFreq[Meta$PoolFreq=="Unknown"] <- "NA"
Meta$PoolFreq[Meta$PoolFreq=="Unspecified"] <- "NA"

#Diet
Meta$Diet<- Meta$"DIET_TYPE"
Meta$Diet[Meta$Diet=="Unknown"] <- "NA"
Meta$Diet[Meta$Diet=="Unspecified"] <- "NA"

#Healthy = No chronic Illness
Meta$Healthy<-Meta$"SUBSET_HEALTHY"
Meta$Healthy[Meta$Healthy=="True"] <- "Healthy"
Meta$Healthy[Meta$Healthy=="False"] <- "Chronic Illness"
Meta$Healthy[Meta$Healthy=="true"] <- "Healthy"
Meta$Healthy[Meta$Healthy=="false"] <- "Chronic Illness"
Meta$Healthy[Meta$Healthy=="Uknown"] <- "NA"
Meta$Healthy[Meta$Healthy=="Unspecified"] <- "NA"

#Recode variables for chronic illnesses to 0-Do not have 1-Dx by provider NA-unknown

Meta$Cancer=Meta$"CANCER"
Meta$Cancer[Meta$Cancer=="Diagnosed by a medical professional doctor, physician assistant"] <- "1"
Meta$Cancer[Meta$Cancer=="Diagnosed by a medical professional (doctor, physician assistant)"] <- "1"
Meta$Cancer[Meta$Cancer=="Diagnosed by an alternative medicine practitioner"] <- "1"
Meta$Cancer[Meta$Cancer=="Self-diagnosed"] <- "0"
Meta$Cancer[Meta$Cancer=="I do not have this condition"] <- "0"
Meta$Cancer[Meta$Cancer=="Unknown"] <- "NA"
Meta$Cancer[Meta$Cancer=="Unspecified"] <- "NA"
Meta$Cancer[Meta$Cancer=="no_data"] <- "NA"


Meta$Kidney=Meta$"KIDNEY_DISEASE"
Meta$Kidney[Meta$Kidney=="Diagnosed by a medical professional doctor, physician assistant"] <- "1"
Meta$Kidney[Meta$Kidney=="Diagnosed by a medical professional (doctor, physician assistant)"] <- "1"
Meta$Kidney[Meta$Kidney=="Diagnosed by an alternative medicine practitioner"] <- "1"
Meta$Kidney[Meta$Kidney=="Self-diagnosed"] <- "0"
Meta$Kidney[Meta$Kidney=="I do not have this condition"] <- "0"
Meta$Kidney[Meta$Kidney=="Unknown"] <- "NA"
Meta$Kidney[Meta$Kidney=="Unspecified"] <- "NA"
Meta$Kidney[Meta$Kidney=="no_data"] <- "NA"


Meta$DM=Meta$"DIABETES"
Meta$DM[Meta$DM=="Diagnosed by a medical professional doctor, physician assistant"] <- "1"
Meta$DM[Meta$DM=="Diagnosed by a medical professional (doctor, physician assistant)"] <- "1"
Meta$DM[Meta$DM=="Diagnosed by an alternative medicine practitioner"] <- "1"
Meta$DM[Meta$DM=="Self-diagnosed"] <- "0"
Meta$DM[Meta$DM=="I do not have this condition"] <- "0"
Meta$DM[Meta$DM=="Unknown"] <- "NA"
Meta$DM[Meta$DM=="Unspecified"] <- "NA"
Meta$DM[Meta$DM=="no_data"] <- "NA"


Meta$Lung=Meta$"LUNG_DISEASE"
Meta$Lung[Meta$Lung=="Diagnosed by a medical professional doctor, physician assistant"] <- "1"
Meta$Lung[Meta$Lung=="Diagnosed by a medical professional (doctor, physician assistant)"] <- "1"
Meta$Lung[Meta$Lung=="Diagnosed by an alternative medicine practitioner"] <- "1"
Meta$Lung[Meta$Lung=="Self-diagnosed"] <- "0"
Meta$Lung[Meta$Lung=="I do not have this condition"] <- "0"
Meta$Lung[Meta$Lung=="Unknown"] <- "NA"
Meta$Lung[Meta$Lung=="Unspecified"] <- "NA"
Meta$Lung[Meta$Lung=="no_data"] <- "NA"


Meta$ASD[Meta$ASD=="Diagnosed by a medical professional doctor, physician assistant"] <- "1"
Meta$ASD[Meta$ASD=="Diagnosed by a medical professional (doctor, physician assistant)"] <- "1"
Meta$ASD[Meta$ASD=="Diagnosed by an alternative medicine practitioner"] <- "1"
Meta$ASD[Meta$ASD=="Self-diagnosed"] <- "0"
Meta$ASD[Meta$ASD=="I do not have this condition"] <- "0"
Meta$ASD[Meta$ASD=="Unknown"] <- "NA"
Meta$ASD[Meta$ASD=="Unspecified"] <- "NA"
Meta$ASD[Meta$ASD=="no_data"] <- "NA"


Meta$IBS[Meta$IBS=="Diagnosed by a medical professional doctor, physician assistant"] <- "1"
Meta$IBS[Meta$IBS=="Diagnosed by a medical professional (doctor, physician assistant)"] <- "1"
Meta$IBS[Meta$IBS=="Diagnosed by an alternative medicine practitioner"] <- "1"
Meta$IBS[Meta$IBS=="Self-diagnosed"] <- "0"
Meta$IBS[Meta$IBS=="I do not have this condition"] <- "0"
Meta$IBS[Meta$IBS=="Unknown"] <- "NA"
Meta$IBS[Meta$IBS=="Unspecified"] <- "NA"
Meta$IBS[Meta$IBS=="no_data"] <- "NA"

 
Meta$IBD[Meta$IBD=="Diagnosed by a medical professional doctor, physician assistant"] <- "1"
Meta$IBD[Meta$IBD=="Diagnosed by a medical professional (doctor, physician assistant)"] <- "1"
Meta$IBD[Meta$IBD=="Diagnosed by an alternative medicine practitioner"] <- "1"
Meta$IBD[Meta$IBD=="Self-diagnosed"] <- "0"
Meta$IBD[Meta$IBD=="I do not have this condition"] <- "0"
Meta$IBD[Meta$IBD=="Unknown"] <- "NA"
Meta$IBD[Meta$IBD=="Unspecified"] <- "NA"
Meta$IBD[Meta$IBD=="no_data"] <- "NA"


Meta$Liver=Meta$"LIVER_DISEASE"
Meta$Liver[Meta$Liver=="Diagnosed by a medical professional doctor, physician assistant"] <- "1"
Meta$Liver[Meta$Liver=="Diagnosed by a medical professional (doctor, physician assistant)"] <- "1"
Meta$Liver[Meta$Liver=="Diagnosed by an alternative medicine practitioner"] <- "1"
Meta$Liver[Meta$Liver=="Self-diagnosed"] <- "0"
Meta$Liver[Meta$Liver=="I do not have this condition"] <- "0"
Meta$Liver[Meta$Liver=="Unknown"] <- "NA"
Meta$Liver[Meta$Liver=="Unspecified"] <- "NA"
Meta$Liver[Meta$Liver=="no_data"] <- "NA"


Meta$CVD=Meta$"CARDIOVASCULAR_DISEASE"
Meta$CVD[Meta$CVD=="Diagnosed by a medical professional doctor, physician assistant"] <- "1"
Meta$CVD[Meta$CVD=="Diagnosed by a medical professional (doctor, physician assistant)"] <- "1"
Meta$CVD[Meta$CVD=="Diagnosed by an alternative medicine practitioner"] <- "1"
Meta$CVD[Meta$CVD=="Self-diagnosed"] <- "0"
Meta$CVD[Meta$CVD=="I do not have this condition"] <- "0"
Meta$CVD[Meta$CVD=="Unknown"] <- "NA"
Meta$CVD[Meta$CVD=="Unspecified"] <- "NA"
Meta$CVD[Meta$CVD=="no_data"] <- "NA"

 
Meta$AutoImm=Meta$"AUTOIMMUNE"
Meta$AutoImm[Meta$AutoImm=="Diagnosed by a medical professional doctor, physician assistant"] <- "1"
Meta$AutoImm[Meta$AutoImm=="Diagnosed by a medical professional (doctor, physician assistant)"] <- "1"
Meta$AutoImm[Meta$AutoImm=="Diagnosed by an alternative medicine practitioner"] <- "1"
Meta$AutoImm[Meta$AutoImm=="Self-diagnosed"] <- "0"
Meta$AutoImm[Meta$AutoImm=="I do not have this condition"] <- "0"
Meta$AutoImm[Meta$AutoImm=="Unknown"] <- "NA"
Meta$AutoImm[Meta$AutoImm=="Unspecified"] <- "NA"
Meta$AutoImm[Meta$AutoImm=="no_data"] <- "NA"


Meta$AD=Meta$"ALZHEIMERS"
Meta$AD[Meta$AD=="Diagnosed by a medical professional doctor, physician assistant"] <- "1"
Meta$AD[Meta$AD=="Diagnosed by a medical professional (doctor, physician assistant)"] <- "1"
Meta$AD[Meta$AD=="Diagnosed by an alternative medicine practitioner"] <- "1"
Meta$AD[Meta$AD=="Self-diagnosed"] <- "0"
Meta$AD[Meta$AD=="I do not have this condition"] <- "0"
Meta$AD[Meta$AD=="Unknown"] <- "NA"
Meta$AD[Meta$AD=="Unspecified"] <- "NA"
Meta$AD[Meta$AD=="no_data"] <- "NA"


Meta$Allergies=Meta$"SEASONAL_ALLERGIES"
Meta$Allergies[Meta$Allergies=="Diagnosed by a medical professional doctor, physician assistant"] <- "1"
Meta$Allergies[Meta$Allergies=="Diagnosed by a medical professional (doctor, physician assistant)"] <- "1"
Meta$Allergies[Meta$Allergies=="Diagnosed by an alternative medicine practitioner"] <- "1"
Meta$Allergies[Meta$Allergies=="Self-diagnosed"] <- "0"
Meta$Allergies[Meta$Allergies=="I do not have this condition"] <- "0"
Meta$Allergies[Meta$Allergies=="Unknown"] <- "NA"
Meta$Allergies[Meta$Allergies=="Unspecified"] <- "NA"
Meta$Allergies[Meta$Allergies=="no_data"] <- "NA"


Meta$Thyroid=Meta$"THYROID"
Meta$Thyroid[Meta$Thyroid=="Diagnosed by a medical professional doctor, physician assistant"] <- "1"
Meta$Thyroid[Meta$Thyroid=="Diagnosed by a medical professional (doctor, physician assistant)"] <- "1"
Meta$Thyroid[Meta$Thyroid=="Diagnosed by an alternative medicine practitioner"] <- "1"
Meta$Thyroid[Meta$Thyroid=="Self-diagnosed"] <- "0"
Meta$Thyroid[Meta$Thyroid=="I do not have this condition"] <- "0"
Meta$Thyroid[Meta$Thyroid=="Unknown"] <- "NA"
Meta$Thyroid[Meta$Thyroid=="Unspecified"] <- "NA"
Meta$Thyroid[Meta$Thyroid=="no_data"] <- "NA"


Meta$Skin=Meta$"SKIN_CONDITION"
Meta$Skin[Meta$Skin=="Diagnosed by a medical professional doctor, physician assistant"] <- "1"
Meta$Skin[Meta$Skin=="Diagnosed by a medical professional (doctor, physician assistant)"] <- "1"
Meta$Skin[Meta$Skin=="Diagnosed by an alternative medicine practitioner"] <- "1"
Meta$Skin[Meta$Skin=="Self-diagnosed"] <- "0"
Meta$Skin[Meta$Skin=="I do not have this condition"] <- "0"
Meta$Skin[Meta$Skin=="Unknown"] <- "NA"
Meta$Skin[Meta$Skin=="Unspecified"] <- "NA"
Meta$Skin[Meta$Skin=="no_data"] <- "NA"

#Subset data to remove children and young adults due to strong association between age and difference in gut microbiome. 
Meta <-subset(Meta, AGE_YEARS>40)

#Make sure it worked
summary(Meta$AGE_YEARS)

```


Look for implausible values in numeric variables that need to be removed from the analysis, and observe distributions. 
```{r}

#Summart Statistics for continuous variables.  

summary(Meta$Age)
summary(Meta$BMI)

#Data Visualization

#Histogram for Age
hist(Meta$Age)
#Not normally distributed, slight skew right. Consider during analysis. 

#Histogram for BMI
hist(Meta$BMI)

#Remove implausible values BMI. Remove any value over 100.Seems mistakes are all in the thousands or higher. New max is 94.48. Need to find method for what is implausible. Obese Class III is >40. Best I have found is any score 3 standard deviations from the mean.<http://apps.who.int/bmi/index.jsp?introPage=intro_3.html> I can also use BMI_CAT for classifications. 
Meta$BMI[Meta$BMI > 100] <- NA

# Re-Do Histogram for BMI and summary
hist(Meta$BMI)
summary(Meta$BMI)

#Scatterplot BMI*AGE to observe for any trends
plot(Meta$BMI, Meta$Age)

#Sumary Statistics for categorical variables (%,SD) observed in tables made in a subsequent step. In my original workflow I performed the table one command first to inspect data by group, rather than running a summary of each indiidual within its own command line. I would like to visualize these in barplots but I was not able to get ggplot to recognize my variable names. Still troubleshooting.
```

Now I will create lists of variables to make analysis easier for the groupings of interest. 
```{r}

#Create a list of all variables of interest
listVars <- c(Age="Age", AgeCat="AGE_CAT", Sex="Sex", Race="Race", BMI="BMI", Education="Education", Antibiotics="Antibiotics", Probiotics="Probiotics", ExFreq="ExFreq", ExLoc="ExLoc", PoolFreq="PoolFreq", Diet="Diet", Healthy="Healthy", Cancer="Cancer", Kidney="Cancer", DM="DM", Lung="Lung", CVD="CVD", AD="AD", Allergies="Allergies", AutoImm="AutoImm", ASD="ASD", IBD="IBD", IBS="IBS", Liver="Liver", Thyroid="Thyroid", Skin="Skin")

#Demographic Variables for Table 1
DemVars <- c(Age="Age", AgeCat="AGE_CAT", Sex="Sex", Race="Race", BMI="BMI", Education="Education", Healthy="Healthy")

#Categorical Demograpic Variables 
CatDemVars <- c(Sex="Sex", Race="Race", Education="Education", Healthy="Healthy")

#Continuous Demograpic Variables
NumDemVars <- c(Age="Age", BMI="BMI")

#Variables to describe lifestyle
LifeVars<- c(Antibiotics="Antibiotics", Probiotics="Probiotics", ExLoc="ExLoc", PoolFreq="PoolFreq", Diet="Diet")

#Specific Illnesses of the sample
Chronic<- c(Cancer="Cancer", Kidney="Cancer", DM="DM", Lung="Lung", CVD="CVD", AD="AD", Allergies="Allergies", AutoImm="AutoImm", ASD="ASD", IBD="IBD", IBS="IBS", Liver="Liver", Thyroid="Thyroid", Skin="Skin")

#Categorical Variables
catVars <- c(Sex="Sex", Race="Race", Education="Education", Antibiotics="Antibiotics", Probiotics="Probiotics", ExFreq="ExFreq", ExLoc="ExLoc", PoolFreq="PoolFreq", Diet="Diet", Healthy="Healthy", Cancer="Cancer", Kidney="Cancer", DM="DM", Lung="Lung", CVD="CVD", AD="AD", Allergies="Allergies", AutoImm="AutoImm", ASD="ASD", IBD="IBD", IBS="IBS", Liver="Liver", Thyroid="Thyroid", Skin="Skin")

#Continuous Variables
contVars <- c(Age="Age", BMI="BMI")
```

Now that the data is reorganized and cleaned it is time to run an initial analysis to (1) see if recoding and grouping worked and (2) look at intitial analyais of the sample data. I will do so by group of self reported Exercise Frequency. 

Now that the data is reorganized and cleaned it is time to run an initial analysis to (1) see if recoding and grouping worked and (2) look at intitial analyais of the sample data. I will do so by group of self reported Exercise Frequency. 
```{r}
#Using "tableone" package

#Figure 1
#Table of demographics by exercise frequency  (Remove "AgeCat" for final figure, but leave in for exploratory analysis here)
Figure1<- CreateTableOne(vars= DemVars, strata = ExFreq, data=Meta, includeNA = FALSE, test = TRUE,testApprox = chisq.test, argsApprox = list(correct = TRUE),
  testExact = fisher.test, argsExact = list(workspace = 2 * 10^5),
  smd = TRUE)
Figure1

#Figure 2
#Table of demographics by exercise frequency 
Figure2<- CreateCatTable(vars= LifeVars, strata = ExFreq, data=Meta, includeNA = FALSE, test = TRUE,testApprox = chisq.test, argsApprox = list(correct = TRUE),
  testExact = fisher.test, argsExact = list(workspace = 2 * 10^5),
  smd = TRUE)
Figure2

#Figure 3
#Table of Chronic Illness Type by exercise frequency 
Figure3<- CreateCatTable(vars= Chronic, strata = ExFreq, data=Meta, includeNA = FALSE, test = TRUE,testApprox = chisq.test, argsApprox = list(correct = TRUE),
  testExact = fisher.test, argsExact = list(workspace = 2 * 10^5),
  smd = TRUE)
Figure3

#Would like to know how to add a column for totals, can't figure out in tableone package, may have to be manual. Also how p-values are actually calculated in this package, what is the input for the chi^2? Also how to omit missing or NA from the analysis. 

#Could not figure this out so I will do it manually in Word. :( I will also insert the tables manually using the word export for this package. It prints shorter and more organized using this function. 

#Recoding of variables above did not change columns in tables for ExFreq. Haven't identified a solution for this yet. Seems to have worked for the variables in the rows. 
```


The following three R chunks have been used to create word document versions of the table. The tableone package seems to print out the tables much better in word than html. Unfortunately these will be inserted into the final knit by hand for aesthetic reasons. However, running the above code will also put them in the document that knits from this file. However, they are much less aesthetically pleasing. 
```{r}
ExFigure1 <- print(Figure1)

# The script
docx( ) %>% 
     addFlexTable(ExFigure1 %>%
     FlexTable(header.cell.props = cellProperties( background.color = "#003366"),
               header.text.props = textBold( color = "white" ),
               add.rownames = TRUE ) %>%
               setZebraStyle( odd = "#DDDDDD", even = "#FFFFFF" ) ) %>%
     writeDoc(file = "~/Desktop/ExFigure1.docx")

```


```{r}
ExFigure2 <- print(Figure2)

# The script
docx( ) %>% 
     addFlexTable(ExFigure2 %>%
     FlexTable(header.cell.props = cellProperties( background.color = "#003366"),
               header.text.props = textBold( color = "white" ),
               add.rownames = TRUE ) %>%
               setZebraStyle( odd = "#DDDDDD", even = "#FFFFFF" ) ) %>%
     writeDoc(file = "~/Desktop/ExFigure2.docx")

```

```{r}
ExFigure3 <- print(Figure3)

# The script
docx( ) %>% 
     addFlexTable(ExFigure3 %>%
     FlexTable(header.cell.props = cellProperties( background.color = "#003366"),
               header.text.props = textBold( color = "white" ),
               add.rownames = TRUE ) %>%
               setZebraStyle( odd = "#DDDDDD", even = "#FFFFFF" ) ) %>%
     writeDoc(file = "~/Desktop/ExFigure3.docx")

```

Although the pvalue is significant, there seems to be a pretty normal distribution of variable characteristics throughout the majority of the groups. However, the sample has a large skew towards a white, well educated population, which I assumed may be the case due to the nature of the project and data collection methods. Submitting your sample to American Gut requires a donation of 80-100 dollars. Due to the demographics there also seems to be a higher amount of people self-reporting routine exercise, rather than the normal distribution of exercise in the population which tends to be lower. This needs to be considered during analysis. Now that the metadata is cleaned I can upload the .biom file and merge the meta data. 


```{r}
##Import OTU file
#Data originally downloaded from <ftp://ftp.microbio.me/AmericanGut/latest>
#Make sure to pull the correct biom file to match the meta data down for analysis. 

# To read in original .biom file use command below 
file_path<- ("~/Desktop/ag_1k_fecal.biom")
dat <- import_biom(file_path)

#To fix error "input string 1 is invalid in this locale" run command "Sys.setlocale(locale="C")" in console and run command chunk again

#Make Phyloseq values for separate pieces of the biom file 
#Taxonomy
taxonomy <- tax_table(dat)
#OTU
OTU_table<- otu_table(dat)

#MetaData
#Create Phyloseq object for sample data so it can be merged
SampleData<- sample_data(Meta)
SD<-sample_data(Meta)

#Import the tree corresponding to this data ### Will Not work
#tree<- tree

#Merge OTU, Taxonomy, and metadata into one phyloseq object called "ps" that can be used for analysis
Sample_data <- (SD)
sam_cov <- as.data.frame(Sample_data)
rownames(sam_cov) <- sam_cov$"#SampleID"
sd <- sample_data(sam_cov)
sample_names(sd) #Previous steps fix error in naming of SampleIds
otus <- otu_table(OTU_table, taxa_are_rows = TRUE)
sample_names(otus)
tt <- tax_table(taxonomy)
ps <- phyloseq(otus, sd, tt) #Step to merge OTU,Taxonomy, and metadata 
ps #View object and make sure it has all 3 parts

```


##Cleaning the OTU table 
First it is important to observe a basic description of the OTU. 
```{r}
#Summary of OTU
ntaxa(ps)
rank_names(ps)


#Summarize sequencing depths 
sdt = data.frame(as(sample_data(ps), "data.frame"),
                 TotalReads = sample_sums(ps), keep.rownames = TRUE)
setNames("rn", "SampleID")
pSeqDepth = ggplot(sdt, aes(TotalReads)) + geom_histogram() + ggtitle("Sequencing Depth")
pSeqDepth

#1000 Reads total
```

Filtering rare taxa is usually a necessary step in this type of data. We will actually perform the filtering elsewhere. For now, just explore the distribution of taxa across the dataset. There are two obvious measures to consider right away: (1) prevalence - the number of samples in which a taxa appears, and (2) total counts - the total number (or proportion) of observations of a taxa across all samples.

```{r}
tdt = data.frame(tax_table(ps),
                 TotalCounts = taxa_sums(ps),
                 OTU = taxa_names(ps))
ggplot(tdt, aes(TotalCounts)) + 
  geom_histogram() + 
  ggtitle("Histogram of Total Counts")

##Seems like an odd number of counts.... doublecheck.
```










## Step 1

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
summary(cars)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
